org.freedesktop.Sdk.Extension.typescript.json

```diff
@@ -0,0 +1,64 @@
+{
+    "id": "org.freedesktop.Sdk.Extension.typescript",
+    "branch": "22.08",
+    "runtime": "org.freedesktop.Sdk",
+    "build-extension": true,
+    "sdk": "org.freedesktop.Sdk",
+    "runtime-version": "22.08",
+    "sdk-extensions": [
+        "org.freedesktop.Sdk.Extension.node18"
+    ],
+    "separate-locales": false,
+    "appstream-compose": false,
+    "build-options": {
+        "prefix": "/usr/lib/sdk/typescript",
+        "append-path": "/usr/lib/sdk/node18/bin",
+        "env" : {
+            "XDG_CACHE_HOME": "/run/build/typescript/flatpak-node/cache",
+            "npm_config_nodedir": "/usr/lib/sdk/node18",
+            "npm_config_cache": "/run/build/typescript/flatpak-node/npm-cache"
+        }
```

Comment:
Don't put module-specific options to the global `build-options`. Move this `env` block to the `typescript` module instead.